gcloud-java-compute/src/test/java/com/google/gcloud/compute/OperationTest.java

```diff
@@ -0,0 +1,489 @@
+    initializeExpectedOperation(3);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.delete(GLOBAL_OPERATION_ID)).andReturn(true);
+    replay(compute);
+    initializeOperation();
+    assertTrue(operation.delete());
+    verify(compute);
+  }
+
+  @Test
+  public void testDeleteFalse() {
+    initializeExpectedOperation(3);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.delete(GLOBAL_OPERATION_ID)).andReturn(false);
+    replay(compute);
+    initializeOperation();
+    assertFalse(operation.delete());
+    verify(compute);
+  }
+
+  @Test
+  public void testExists_True() throws Exception {
+    initializeExpectedOperation(3);
+    Compute.OperationOption[] expectedOptions = {Compute.OperationOption.fields()};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(globalOperation);
+    replay(compute);
+    initializeOperation();
+    assertTrue(operation.exists());
+    verify(compute);
+  }
+
+  @Test
+  public void testExists_False() throws Exception {
+    initializeExpectedOperation(3);
+    Compute.OperationOption[] expectedOptions = {Compute.OperationOption.fields()};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(null);
+    replay(compute);
+    initializeOperation();
+    assertFalse(operation.exists());
+    verify(compute);
+  }
+
+  @Test
+  public void testIsDone_True() throws Exception {
+    initializeExpectedOperation(3);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(globalOperation);
+    replay(compute);
+    initializeOperation();
+    assertTrue(operation.isDone());
+    verify(compute);
+  }
+
+  @Test
+  public void testIsDone_False() throws Exception {
+    initializeExpectedOperation(4);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(
+        Operation.fromPb(serviceMockReturnsOptions, globalOperation.toPb().setStatus("PENDING")));
+    replay(compute);
+    initializeOperation();
+    assertFalse(operation.isDone());
+    verify(compute);
+  }
+  @Test
+  public void testIsDone_NotExists() throws Exception {
+    initializeExpectedOperation(3);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(globalOperation);
+    replay(compute);
+    initializeOperation();
+    assertTrue(operation.isDone());
```

Comment:
Yes, in fact the method should return `null`, not `globalOperation`. Copy-paste error, nice catch!